### PR TITLE
Tiny MCE plugin overrides custom settings

### PR DESCRIPTION
**Pull request by DerinMavi**
See #14033

### Summary of Changes
Issue reported here: #9812, #8591, #5820, #11851 and #8106
Some of the issues were closed because of the pull requests that came but later it was found out either they do not work or they are wrong(they modify default values).

This change fixes the problem while keeping the default values.

### Testing Instructions
Please test all 3 use cases:

1) Use the following code to display a tinymce editor:
`
$editor = JEditor::getInstance('tinymce');
echo $editor->display('description', "", '250', '50', '60', '5', false);`

2) Add a new article from backend of the site, try to edit this xml code https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_content/models/forms/article.xml#L27 , add width and height attributes (see https://docs.joomla.org/Editor_form_field_type for document about editor field type). For example, change it to;
`<field name="articletext" type="editor"
			label="COM_CONTENT_FIELD_ARTICLETEXT_LABEL" description="COM_CONTENT_FIELD_ARTICLETEXT_DESC"
			filter="JComponentHelper::filterText" buttons="true"  height="50" width="250" />`

3) In the joomla backend go to extensions -> plugins -> tinymce editor -> advanced settings and modify width & height

### Expected result
Tinymce editor has height of 50pixels and width of 250pixels
### Actual result
Tinymce editor has height of 550pixels

### CODE REVIEW

File: plugins/editors/tinymce/tinymce.php [hunk 1]
```diff
@@ -405,13 +405,23 @@ public function onDisplay(
...
-			$html_width = '';
+			$html_height = $height;
+		}
+		else
+		{
+			$html_height = $this->params->get('html_height', '550');
+		}
+		
+		if ($width)
+		{
```
Comment (SharkyKZ) [inline plugins/editors/tinymce/tinymce.php]:
```suggestion
		{
```

File: plugins/editors/tinymce/tinymce.php [hunk 2]
```diff
@@ -405,13 +405,23 @@ public function onDisplay(
 			$extended_elements = trim($levelParams->get('extended_elements', ''));
 			$valid_elements    = trim($levelParams->get('valid_elements', ''));
 		}
-
-		$html_height = $this->params->get('html_height', '550');
-		$html_width  = $this->params->get('html_width', '');
-
-		if ($html_width == 750)
+
```
Comment (SharkyKZ) [inline plugins/editors/tinymce/tinymce.php]:
```suggestion

```

**Comment by schnuti** [comment]
I have tested this item :white_check_mark: successfully on 78de832c1fa8a6b47c31a57616662946c923981e<br /><br />Tested as described + edited the code in J4.0 and tested with the same positive result.
  <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26755">issues.joomla.org/tracker/joomla-cms/26755</a>.</sub>

**Comment by schnuti** [comment]
One problem remains: Width and height seems not to be per instance. i.e. using more than 1 editor field on a page can not have different width and/or height.
For another issue!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/26755">issues.joomla.org/tracker/joomla-cms/26755</a>.</sub>

**Comment by schnuti** [comment]
@DerinMavi 
This should fix the instance issue - to add before line 151 in J4.0. 
https://github.com/joomla/joomla-cms/blob/1ec557f6e8cfceec031c58cc4280d05b4f3289cc/plugins/editors/tinymce/tinymce.php#L151
and before line 221 in J3
https://github.com/joomla/joomla-cms/blob/946aa6130209a0e95dd2aaffdb4e197c84c6a7c1/plugins/editors/tinymce/tinymce.php#L221
  
     // Set custom width and height per instance
    if ($width) {
        $options['tinyMCE'][$fieldName]['width']    = $width;
    }
    
    if ($height) {
        $options['tinyMCE'][$fieldName]['height']    = $height;
    }

**Comment by hakanara** [comment]
Hello, I am writing from the future, from the date of February 3rd, 2020. I see that the issue is open and we cannot still overwrite the existing height of the editor. Anything planned? Kind regards.

**Comment by TLWebdesign** [comment]
Anything new on this? Running into this issue now too on a component i built where i want a small height for the editor.

**Comment by zero-24** [comment]
Seems just tests are missing so please test that this change here works and mark an successfull test. After two successfull tests this gets into RTC and than merged into the core. :-)

**Comment by TLWebdesign** [comment]
I tested but nr3 did not change anything for me in my custom component where i have multiple editors on screen and also not in the article content editor. I set height in the plugin settings to 900 but it stays at 500. 

Also the affected lines are different nowadays. The instance issue from @schnuti worked just fine!

**Comment by zero-24** [comment]
hmm seems we have a mix of different default values.

- The default value in the editor plugin field
https://github.com/joomla/joomla-cms/blob/3.9.21/libraries/src/Form/Field/EditorField.php#L208-L209
  - The value found here is passed as height and width of the textarea of the tinymce field: https://github.com/joomla/joomla-cms/blob/3.9.21/plugins/editors/tinymce/tinymce.php#L191-L192
- The other default value is the html height and width. This is configured from the plugin:
https://github.com/joomla/joomla-cms/blob/3.9.21/plugins/editors/tinymce/tinymce.php#L411-L412

I have not yet understood why there is a different between html_height / html_width that come from the plugin options and height / width that come from the filed options.

In the other editor plugin (codemirror) it seems that height / width is ignored altogether as this seems to be the last mention of it in the plugin: https://github.com/joomla/joomla-cms/blob/3.9.21/plugins/editors/codemirror/codemirror.php#L191-L192

**Comment by laoneo** [comment]
I'v tested this pr and the height got completely ignored. Also test 3 didn't work as mentioned by @TLWebdesign. I made a new pr #37603 which should fix all situations mentioned here. So I'm closing this one. Please give #37603 a test or leave any feedback so we can get it fixed finally. Thanks for your participation.

**Pull request closed without merging**